README.md

```diff
@@ -1,16 +1,148 @@
+
+
+#### Trace
+Trace component is designed to support distributed tracing (see dapper paper). This component 
+allows collecting and recording in memory trace data, tracks active requests, and keeps local 
+samples for interesting requests.
+
+The key elements of the API can be broken down as:
+* A Span represents a single operation within a trace. Spans can be nested to form a trace tree. 
+* Libraries must allow users to record tracing events for a span (attributes, annotations, links, 
+etc.).
+* Spans carried in the Generic Context. Libraries MUST provide a means of manipulating the Span in
+the context, including getting the current Span, and replacing the current Span.
+* Support dynamically control of the global configuration (e.g. trace sampling rate/probability). 
+* Libraries MUST provide a means of dynamically controlling the trace global config at runtime.
+* Libraries SHOULD keep track of active spans and in memory samples based on latency/errors and 
+offer ways to access the data.
+* Because “trace context” must also be propagated between process, library MUST offer the 
+functionality that allows any transport (e.g RPC, HTTP, etc.) systems to encode/decode the “trace
+context” for placement on the wire.
+
+##### Links
+* Trace API is described [here][TraceAPI]
+* Data model is defined [here][TraceDataModel]
+
+#### Tags
+Tags are used by the Stats API to break down measurements by arbitrary metadata set in the 
+current process or propagated from a remote caller (tags are propagated through the Generic 
+Context subsystem inside a process and between process by any transport (e.g RPC, HTTP, etc.) 
+systems).
+
+The key elements of the Tags API are:
+* A tag: this is a key-value pair, where the key is a string, and the value can be one of a 64-bit
+integer, a boolean, or a string. The API allows for creating, modifying and querying objects 
+representing a tag value.
+* A set of tags (with unique keys) carried in the Generic Context. Libraries MUST provide a means 
+of manipulating the tags in the context, including adding new tags, replacing tag values, deleting
+tags, and querying the current value for a given tag key. 
+* Because tags must also be propagated between process, library MUST offer the functionality that 
+allows RPC systems to encode/decode the set of tags for placement on the wire.
+
+##### Links
+* TODO: Add links to API definition and data model.
+
+#### Stats
+The Stats API is designed to record measurements, dynamically break them down by 
+application-defined tags, and aggregate those measurements in user-defined ways. It is designed 
+to offer multiple types of aggregation (e.g. Distributions) and be efficient (all measurement 
+processing is done as a background activity); aggregating data enables reducing the overhead of 
+uploading data, while also allowing applications direct access to stats for (e.g.) load-balancing
```

Comment:
for e.g. is strange. Use for example instead.

Reply:
Explain how a load balancer can use this data. You are talking about a specific case I already know but for someone who is reading this for the first time, it is hard to follow.

Reply:
Done. for example.

Reply:
Failed to explain in few words so I decided to remove this extra example. we can add more examples later.